packages/commonwealth/client/scripts/views/components/Comments/CommentEditor/CommentEditor.tsx

```diff
@@ -0,0 +1,112 @@
+import BN from 'bn.js';
+import { weiToTokens, getDecimals } from 'helpers';
+import { ContentType } from 'types';
+import type { DeltaStatic } from 'quill';
+
+type CommentEditorProps = {
+  parentType: ContentType;
+  canComment: boolean;
+  handleSubmitComment: () => void;
+  errorMsg: string;
+  contentDelta: DeltaStatic;
+  setContentDelta: React.Dispatch<React.SetStateAction<DeltaStatic>>;
+  tokenPostingThreshold: BN;
+  activeTopic: Topic;
+  userBalance: BN;
+  disabled: boolean;
+  cancel: (e: any) => void;
+  isAdmin: boolean;
+  author: Account;
+  editorValue: string;
+  shouldFocus: boolean;
+};
+
+export const CommentEditor = ({
+  parentType,
+  canComment,
+  handleSubmitComment,
+  errorMsg,
+  contentDelta,
+  setContentDelta,
+  tokenPostingThreshold,
+  activeTopic,
+  userBalance,
+  disabled,
+  cancel,
+  isAdmin,
+  author,
+  editorValue,
+  shouldFocus,
+}: CommentEditorProps ) => {
+
+  const decimals = getDecimals(app.chain);
+
+  return (
+    <div className="CreateComment">
+      <div className="attribution-row">
+        <div className="attribution-left-content">
+          <CWText type="caption">
+            {parentType === ContentType.Comment ? 'Reply as' : 'Comment as'}
+          </CWText>
+          <CWText
+            type="caption"
+            fontWeight="medium"
+            className={clsx('user-link-text', { disabled: !canComment })}
+          >
+            <User
+              userAddress={author?.address}
+              userChainId={author?.chain.id}
+              shouldHideAvatar
+              shouldLinkProfile
+            />
+          </CWText>
+        </div>
+        {errorMsg && <CWValidationText message={errorMsg} status="failure" />}
+      </div>
+      <ReactQuillEditor
+        className="editor"
+        contentDelta={contentDelta}
+        setContentDelta={setContentDelta}
+        isDisabled={!canComment}
+        tooltipLabel="Join community to comment"
+        shouldFocus={shouldFocus}
+      />
+      {tokenPostingThreshold && tokenPostingThreshold.gt(new BN(0)) && (
+        <CWText className="token-req-text">
+          Commenting in {activeTopic?.name} requires{' '}
+          {weiToTokens(tokenPostingThreshold.toString(), decimals)}{' '}
+          {app.chain.meta.default_symbol}.{' '}
+          {userBalance && (
+            <>
+              You have {weiToTokens(userBalance.toString(), decimals)}{' '}
+              {app.chain.meta.default_symbol}.
+            </>
+          )}
+        </CWText>
+      )}
+      <div className="form-bottom">
+        <div className="form-buttons">
+          {editorValue.length > 0 && (
+            <CWButton buttonType="tertiary" onClick={cancel} label="Cancel" />
+          )}
+          <CWButton
+            buttonWidth="wide"
+            disabled={disabled && !isAdmin}
+            onClick={handleSubmitComment}
+            label="Submit"
+          />
+        </div>
+      </div>
+    </div>
+  )
+}
```

Comment:
prettier: file is not formatted correctly

Reply:
Done